xbmc/filesystem/AddonsDirectory.cpp

```diff
@@ -825,18 +824,21 @@ void CAddonsDirectory::GenerateAddonListing(const CURL& path,
                                                                     addon->Version());
     bool disabled = CServiceBroker::GetAddonMgr().IsAddonDisabled(addon->ID());
 
-    bool isUpdate =
-        std::any_of(updates.begin(), updates.end(), [&](const std::shared_ptr<IAddon>& i) {
-          return i->ID() == addon->ID() && i->Origin() == addon->Origin() &&
-                 i->Version() == addon->Version();
-        });
-
-    bool hasUpdate =
-        alwaysShowUpdateIcon ||
-        std::any_of(outdated.begin(), outdated.end(), [&](const std::shared_ptr<IAddon>& i) {
-          return i->ID() == addon->ID() && i->Origin() == addon->Origin() &&
-                 i->Version() == addon->Version();
-        });
+    auto CheckOutdatedOrUpdate = [&](bool checkOutdated) -> bool {
+      const auto& mapEntry = outdatedWithUpdates.find(addon->ID());
+      if (mapEntry != outdatedWithUpdates.end())
+      {
+        const std::shared_ptr<IAddon>& checkedObject =
+            checkOutdated ? mapEntry->second.outdated : mapEntry->second.update;
+
+        return (checkedObject->Origin() == addon->Origin() &&
+                checkedObject->Version() == addon->Version());
+      }
+      return false;
+    };
```

Comment:
I'm not sure what is the accepted way to do this, declare `auto CheckOutdatedOrUpdate =` or to not use auto and instead `std::function<bool(bool)> CheckOutdatedOrUpdate =`.

I'm not saying it's wrong I just have not seen a function declared like this previously!

@ksooo, do you know? Or is either way good? When I scan this code I see a variable definition until I look deeper.